### ISSUE TITLE
Pkg.init() fails (Not due to git insteadof https)

### ISSUE THREAD

**Issue by jne555** [JuliaLang/julia#14211]
I am running windows 7 64-bit and recently downloaded Julia 0.4.1 (commit cbe1bee from 8/11/15 at 10:33). I launch Julia from within the 0.4.1 install and run Pkg.init() and it fails on the cloning of Metadata with the following error:

julia> Pkg.init()
09:42:30.866866 git.c:348               trace: built-in: git 'version'
INFO: Initializing package repository C:\Users\Joel\.julia\v0.4
INFO: Cloning METADATA from git://github.com/JuliaLang/METADATA.jl
09:42:31.147666 git.c:348               trace: built-in: git 'clone' '-q' '-b' 'metadata-v2' 'git://github.com/JuliaLang/METADATA.jl' 'METADATA'
09:42:31.334867 run-command.c:335       trace: run_command: 'git-remote-https' 'origin' 'https://github.com/JuliaLang/METADATA.jl'
ERROR: failed process: Process(`git clone -q -b metadata-v2 git://github.com/JuliaLang/METADATA.jl METADATA`, ProcessExited(128)) [128]
 in run at process.jl:531

I originally tried with my current installed version of Git I had been using for some time and thinking maybe it was a Git issue I un-installed and re-installed with the newest version of Git (Still was getting exact same error.) So, I decided to un-install Git and see if the version of Git installed with Julia would possibly fix the issue. But, still same error.

julia> run(`git --version`)
09:43:03.801336 git.c:348               trace: built-in: git 'version'
git version 2.6.1.windows.1

Here is the versioninfo() of my environment.

julia> versioninfo()
Julia Version 0.4.1
Commit cbe1bee* (2015-11-08 10:33 UTC)
Platform Info:
  System: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-3610QM CPU @ 2.30GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.3

I was previously using version 0.3.8 and if I delete my ~/.julia folder and launch Julia version 0.3.8 and run Pkg.init() it completes just fine. I've tried re-installing Julia 0.4.1 and re-installing Git... I'm starting to run out of ideas what could be causing the issue.

Joel

**Comment by tkelman** [comment]
So if you run that same git clone command from git-cmd or git-bash of git 2.6.x, does it work or not?

**Comment by jne555** [comment]
I can run the command 'git clone -q -b metadata-v2 git://github.com/JuliaLang/METADATA.jl METADATA and it will clone the repository'. However, if I go into the Julia-0.4.1\Git\Bin and run the command '.\git clone -q -b metadata-v2 git://github.com/JuliaLang/METADATA.jl' it fails with the following error:

![image](https://cloud.githubusercontent.com/assets/6857860/11509578/ce9a1698-982c-11e5-9419-e26b23f773a5.png)

So, I was thinking it was using my installed version of Git but must actually be using the one within the Julia installation. The error seems to be related to OpenSSL and so I'm trying to figure out how to remove and install the correct version of OpenSSL to get this working.

**Comment by tkelman** [comment]
that shouldn't be necessary. do you have some other version of openssl on your path? better to fix your path so it's not always there. the git launcher exe in Git\bin should handle setting paths correctly for its dependencies.

**Comment by jne555** [comment]
Yeah, I was thinking to dowload OpenSSL binaries might fix it and I added a reference in my path to them.

C:\Users\Joel\Documents\openssl-0.9.8k_X64\bin

But, that didn't work so I removed the reference for OpenSSL and still same error. There is ssleay32.dll in my system32 folder so at some point an application installed the dll though I'm not certain which one.

**Comment by tkelman** [comment]
Can you check the creation / modification times and try to cross reference what you might have installed around that time? Try moving it out of the way and see if anything breaks? Installing random dlls into system32 is really bad practice.

**Comment by jne555** [comment]
Yeah, I checked the create time on the dll and checked my programs to see if anything matched, but nothing was really within a week of it. I also tried to un-register the dll and it claimed "no entry point"... so I went ahead and deleted it.

Once I did that the Pkg.init() finished successfully. Wish I knew what caused the original conflict, but at least things are good now. Thanks for the help!!

Joel
